### ISSUE TITLE
no-invalid-this error is suppressed in node environment

### ISSUE THREAD

**Issue by mhipszki** [eslint/eslint#3961]
**The version of ESLint you are using (run eslint -v)**

1.5.1

**What you did (the source code and ESLint configuration)**

source code

```js
// some Node.js module's code
'use strict';

this.a = 3;
```

eslint config

```json
{
    "extends": "eslint:recommended",
    "env": { "node": true },
    "rules": {
        "no-invalid-this": 2
    }
}
```

**The actual ESLint output complete with numbers**

*nothing*

**What you expected to happen instead**

Expected ESLint to report `no-invalid-this` error, but instead got nothing.

`this` refers to the module object itself in a context of a Node module, so it won't be *undefined* in any case.

However I would not encourage developers to use `this.a = 3;` in Node modules instead of `exports.a = 3;`  as the latter one is more explicit, however it might be just my personal preference?

Node modules are *cached* after loading, which actually advocating the "singleton pattern" in a way, and again it might be my own preference, but I would not encourage developers to do that as a habit.

See https://nodejs.org/api/modules.html#modules_caching

**Proposal**: an option to warn developers not to use `this` in Node modules' root scope.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by michaelficarra** [comment]
This is simply a bug in `no-invalid-this`. It should report top-level `this`, but is likely getting tripped up by the implicit top-level function wrapper in node programs.

**Comment by mysticatea** [comment]
Thanks for the report.
It's a bug of `no-invalid-this`.
I will fix it.
